onedrive_service/src/onedrive_service/authentication/one_drive_client.py

```diff
@@ -0,0 +1,43 @@
+import onedrivesdk
+from onedrivesdk.helpers import GetAuthCodeServer
+
+from .session import Session
+from .base_client import BaseOneDriveClient
+
+
+API_BASE_URL = "https://api.onedrive.com/v1.0/"
+
+
+class OneDriveClient(BaseOneDriveClient):
+    """ API client for OneDrive for personal use.
+    """
+
+    def __init__(self, config: dict) -> None:
+        """ Basic initialization.
+        """
+
+        self._config = config
+        client_id = config["auth"]["client_id"]
+        scopes = config["auth"]["scopes"]
+
+        http_provider = onedrivesdk.HttpProvider()
+        auth_provider = onedrivesdk.AuthProvider(http_provider, client_id, scopes, session_type=Session)
+
+        super().__init__(API_BASE_URL, auth_provider, http_provider)
+
+    def authenticate(self, client_secret=None, redirect_uri="http://localhost:8080/", auth_code=None):
+        """ Authenticate process for OneDrive for personal use.
+        """
+
+        client_secret = client_secret if client_secret is not None else self._config["auth"]["client_secret"]
+
+        if self.check_saved_session_exists():
+            self.auth_provider.load_session()
+
+        else:
+            if auth_code is None:
```

Comment:
if not auth_code

Reply:
Explicit is better.

> Comparisons to singletons like None should always be done with is or is not, never the equality operators.
> Also, beware of writing if x when you really mean if x is not None -- e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be false in a boolean context!

https://www.python.org/dev/peps/pep-0008/